From 7b25624bafbbeadd30051700c0a5f74f9bc69b5a Mon Sep 17 00:00:00 2001
From: Haim Boot <hayim@marvell.com>
Date: Wed, 4 May 2016 18:39:59 +0300
Subject: [PATCH 144/239] ddr: ble: a7040-rz: add parameter file to allow board
 specific configuation

- till now a7040-rz & a70x0 shared the same file and therefore the same
  confiuration
- since a70x0 was limited to 2GB it also limited a7040-rz which has 4GB.
- this patch allow configure a7040-rz dram to 4GB (3.8GB)

Change-Id: I35bfba4c7d58173e3c5ee78a5a1b7256dbba0b64
Signed-off-by: Haim Boot <hayim@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29507
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Victor Axelrod <victora@marvell.com>
---
 plat/marvell/a8k/a7040_rz/board/dram_port.c | 75 +++++++++++++++++++++++++++++
 plat/marvell/a8k/a7040_rz/platform.mk       | 10 ++--
 2 files changed, 81 insertions(+), 4 deletions(-)
 create mode 100644 plat/marvell/a8k/a7040_rz/board/dram_port.c

diff --git a/plat/marvell/a8k/a7040_rz/board/dram_port.c b/plat/marvell/a8k/a7040_rz/board/dram_port.c
new file mode 100644
index 0000000..c9b8346
--- /dev/null
+++ b/plat/marvell/a8k/a7040_rz/board/dram_port.c
@@ -0,0 +1,75 @@
+/*
+* ***************************************************************************
+* Copyright (C) 2016 Marvell International Ltd.
+* ***************************************************************************
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+*
+* Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following disclaimer in the documentation
+* and/or other materials provided with the distribution.
+*
+* Neither the name of Marvell nor the names of its contributors may be used
+* to endorse or promote products derived from this software without specific
+* prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+* OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+***************************************************************************
+*/
+
+#include <arch_helpers.h>
+#include <plat_marvell.h>
+#include <debug.h>
+#include <dram_if.h>
+
+struct dram_config dram_cfg = {
+	.iface_mask	= 0x1,
+	.printf		= printf,
+
+	/* Interface configuration */
+	.iface[0] = {
+		.mac_base = (void *)MVEBU_DRAM_MAC_BASE,
+		.phy_base = (void *)MVEBU_DRAM_PHY_BASE,
+		.ecc_enabled	= 0,
+		.cs_count	= 1,
+		.freq_mhz	= 650,
+
+		/* Bus configuration */
+		.bus[0] = {
+			.bus_width = 32,
+			.size_mb   = 0x1000,
+		},
+	},
+};
+
+/* This function may modify the default DRAM parameters
+ * based on information recieved from SPD or bootloader
+ * configuration located on non volatile storage */
+int update_dram_info(struct dram_config *cfg)
+{
+	NOTICE("Gathering DRAM information\n");
+	return 0;
+}
+
+void *plat_get_dram_data(void)
+{
+	/* Update DRAM for dynamic platforms */
+	update_dram_info(&dram_cfg);
+
+	return &dram_cfg;
+}
diff --git a/plat/marvell/a8k/a7040_rz/platform.mk b/plat/marvell/a8k/a7040_rz/platform.mk
index d8001a0..87f7ae9 100644
--- a/plat/marvell/a8k/a7040_rz/platform.mk
+++ b/plat/marvell/a8k/a7040_rz/platform.mk
@@ -63,12 +63,14 @@ PLAT_INCLUDES		:=	-I$(PLAT_FAMILY_BASE)/$(PLAT_SRC)		\
 				${ATF_INCLUDES}
 
 PLAT_BL_COMMON_SOURCES	:=	$(PLAT_SRC_BASE)/aarch64/a7040_rz_common.c	\
-				drivers/console/console.S							\
+				drivers/console/console.S			\
 				drivers/ti/uart/16550_console.S
 
-BLE_SOURCES		:=	$(PLAT_SRC_BASE)/board/dram_port.c		\
-				plat/marvell/common/sys_info.c			\
-				plat/marvell/common/dram_setup.c
+BLE_PORTING_SOURCES	:=	$(PLAT_FAMILY_BASE)/$(PLAT)/board/dram_port.c
+
+BLE_SOURCES		:=	plat/marvell/common/sys_info.c			\
+				plat/marvell/common/dram_setup.c		\
+				$(BLE_PORTING_SOURCES)
 
 BL1_SOURCES		+=	$(PLAT_SRC_BASE)/aarch64/plat_helpers.S	\
 				lib/cpus/aarch64/cortex_a72.S
-- 
1.9.1

